### PR TITLE
reflect: Implement CLI

### PR DESCRIPTION
**Pull request by juliaogris**
Implement reflect CLI to query gRPC reflection API as

	reflect [services|filename|symbol|extensions|extension] ...

For further details see README.md and `reflect --help`.

Add golden test comparing expected again real file output for json,
bas64 and binary formats.

Next up: Add `echo2` service for proto2 tests.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,9 +8,7 @@ install from source with
 
 	go install github.com/juliaogris/reflect@latest
 
-## Development
-
-### Pre-requistes
+## Development [re-requistes
```
Comment (camh-) [inline README.md]: s/[/p/

File: main.go [hunk 2]
```diff
@@ -1,30 +1,261 @@
...
+	Version    kong.VersionFlag `short:"V" help:"Print version information" group:"Other:"`
+	Services   servicesCmd      `cmd:"" help:"Call list_services"`
+	Symbol     symbolCmd        `cmd:"" help:"Call file_containing_symbol"`
+	Filename   filenameCmd      `cmd:"" help:"Call file_by_filename"`
+	Extension  extensionCmd     `cmd:"" help:"Call file_containing_extension"`
+	Extensions extensionsCmd    `cmd:"" help:"Call all_extension_numbers_of_type"`
+}
+
+type (
+	servicesCmd struct{}
```
Comment (camh-) [inline main.go]: a little bit odd that you've grouped these two types but no others.

File: main.go [hunk 3]
```diff
@@ -1,30 +1,261 @@
...
+		},
+	}
+	resp, err := send(stream, req)
+	if err != nil {
+		return err
+	}
+	return printProto(g.out, resp, g.Format)
+}
+
+func (e *extensionsCmd) Run(g globals) error {
```
Comment (camh-) [inline main.go:124]: All of the `Run` methods seem to be identical except for the value of `MessageRequest`. Any reason not to factor it out and just create and pass the `MessageRequest` in each `Run` method?

File: main.go (hunk 2 above)
Comment (juliaogris) [reply main.go]: yes, oups, oversight, fixed.

File: main.go (hunk 3 above)
Comment (juliaogris) [reply main.go:124]: yes good idea.

**Review by camh- — APPROVED** [review]
LGTM :frog:

**Pull request merged**